docs/sql/ad-hoc.md

```diff
@@ -1,29 +1,55 @@
 # Ad-hoc SQL queries
 
-You can run ad-hoc SQL queries on a running pipeline.
-Unlike SQL programs that define pipelines and are evaluated incrementally, ad-hoc queries are evaluated using a batch engine against the current snapshot of pipeline's tables and views. Ad-hoc queries can be executed both when the pipeline is running or paused.
+You can run ad-hoc SQL queries on a running or paused pipeline. Unlike Feldera SQL programs that define pipelines and
+are evaluated incrementally, ad-hoc queries are evaluated in batch-mode,
+using the [datafusion engine](https://datafusion.apache.org).
 
-Ad-hoc queries provide a very cheap way to query the state of a materialized view. They are designed to aid development and debugging, so you need to be aware of their limitations to avoid potential confusion.
+Ad-hoc queries provide a cheap way to query the state of a materialized view. They are designed to aid
+development and debugging, so you need to be aware of their limitations to avoid potential confusion.
 
 ## Limitations
 
-:::note
+:::info
 
 Tables and views are only accessible to `SELECT` ad-hoc queries if they are declared as [materialized](/sql/materialized).
 
 :::
 
-Feldera does not guarantee consistency between SQL dialects of Feldera programs and ad-hoc queries as the former is used in an incremental view maintenance (IVM) context and the latter is not, so more complex ad-hoc queries may produce different results than an SQL program.
+As of now, there are differences between the SQL dialects of Feldera SQL programs and ad-hoc queries.
+This is because they use different engines (Apache Calcite for Feldera SQL vs. Apache Datafusion for ad-hoc queries).
+See below for some examples.
 
-Currently, only `SELECT` and `INSERT` statements are supported. You can not create or alter tables and views using ad-hoc SQL.
+Currently, only `SELECT` and `INSERT` statements are supported. You can not create or alter tables and views using
+ad-hoc SQL.
+
+### Differences between Feldera SQL and Ad-hoc Queries
+
+Ad-hoc queries may produce slightly different results compared to what Feldera outputs due to differences in query
```

Comment:
The first two sentences contradict each other. I think it's ok to say that they produce consistent results on the common subset with some exceptions (we might discover more exceptions in the future).